modules/core/src/main/scala/doobie/util/fragments.scala

```diff
@@ -20,41 +20,41 @@ object fragments {
   def notIn[F[_]: Reducible, A: Put](f: Fragment, fs: F[A]): Fragment =
     fs.toList.map(a => fr0"$a").foldSmash1(f ++ fr0"NOT IN (", fr",", fr")")
 
-  /** Returns `(f1) AND (f2) AND ... (fn)`. */
-  def and(fs: Fragment*): Fragment =
-    fs.toList.map(parentheses).intercalate(fr"AND")
+  /** Returns `(f1 AND f2 AND ... fn)`, or nothing for empty `fs`. */
+  def and(fs: Fragment*): Option[Fragment] =
+    if (fs.isEmpty) None else Some(parentheses(fs.toList.intercalate(fr"AND")))
```

Comment:
We want each fragment parenthesized here; i..e, `(a) AND (b) AND (c)` so things like `(a OR b) AND (c or d)` don't turn into `a OR b AND c OR d` which will be parsed as ((a OR b) AND c) or d`.

I wonder if we want to return `true` when `fs` is empty, and `false` for `or` ... ?

Reply:
> each fragment parenthesized here; i..e, `(a) AND (b) AND (c)`

that would be tackling the issue from the wrong end. `and` should itself return a "single" expression, ie something parenthesized
so that `a OR b AND c OR d` is not possible to construct. `and(or(a, b), or(c, d))` will always create `((a OR b) AND (c or d))`
see the newly added tests to see how this makes these functions (`and`, `or`,...) composable and how the current approach would fail here.

Reply:
Ok, that makes sense. I will look at the tests. I'm a little concerned that this will quietly break existing code so we should add a release note about it.

Reply:
thanks! this (we could even consider it a bug) has bitten me. I tried to come up with a solution which wouldn't be as disruptive, but couldn't.
please ask any question you might have about my proposed fix